app/questionnaire/navigation.py

```diff
@@ -200,9 +162,58 @@ def _generate_link_names(self, label_answer_ids):
 
         return link_names
 
-    @staticmethod
-    def _get_visible_groups_for_section(section):
-        return (
-            group for group in section['groups']
-            if not group.get('hide_in_navigation')
+    def _get_location_for_section(self, section, groups):
+        section_completeness = self.completeness.get_state_for_section(section)
+        first_incomplete = self.completeness.get_first_incomplete_location_in_section(section)
+        first_group, last_group = groups[0], groups[-1]
+
+        return self._get_location_for_completeness(
+            section_completeness,
+            first_group,
+            last_group,
+            first_group['blocks'][0],
+            last_group['blocks'][-1],
+            first_incomplete
         )
+
+    def _get_location_for_group(self, group, group_instance=None):
+        """Gets the location for a link used for a repeating group
+        """
+        group_completeness = self.completeness.get_state_for_group(group, group_instance=group_instance)
+        first_incomplete = self.completeness.get_first_incomplete_location_in_group(group, group_instance=group_instance)
+
+        return self._get_location_for_completeness(
+            group_completeness,
+            group,
+            group,
```

Comment:
Should this be reapeated_group or something?

Reply:
You mean because the same value is passed twice? No, in this case that's intentional. This method gets a link for a repeating group, all of the possible locations are within the same group so the value passed for `first_group` and `last_group` are both the same.

Reply:
Ahhh ok yes that was it, in which case this is G2G